src/components/Navigator/NavigatorCard.vue

```diff
@@ -1014,37 +1018,65 @@ export default {
-        // try to match if any of the `siblings`,`children` or any of the `parents`,
-        // match the current open item
-        const matchingItem = [...children, ...siblings, ...parents]
-          .find(child => child.path === lastActivePathItem);
+        // try to match current open item in its surroundings, starting with active item
+        if (this.matchSurroundingItems(this.activeUID, lastActivePathItem)) return;
 
-        // set the match as an active item
-        if (matchingItem) {
-          this.setActiveUID(matchingItem.uid);
-          return;
+        if (this.isSpecificOverload) {
+          // if no match, try again to match with generic item
+          // Needed for continuing to highlight current generic page
+          // when selecting an overload from dropdown that's also specifically curated in elsewhere
+          const genericItem = lastActivePathItem.slice(0, lastActivePathItem.lastIndexOf('-'));
+          if (this.matchSurroundingItems(this.activeUID, genericItem)) return;
         }
       }
-      // There is no match to base upon, so we need to search
-      // across the activePath for the active item.
+      // There is no match to base upon, so we need to search the whole tree
+      // by matching each level of the hierachy in activePath
       const activePathChildren = this.pathsToFlatChildren(activePath);
-      // if there are items, set the new active UID
+      // if there are items, set new active UID
       if (activePathChildren.length) {
-        this.setActiveUID(activePathChildren[activePathChildren.length - 1].uid);
+        const lastChildrenUID = last(activePathChildren).uid;
+
+        if (last(activePathChildren).path !== lastActivePathItem && this.isSpecificOverload) {
+          // if item is not found in the tree and its a specific overloaded symbol page
+          // try to match with its generics page instead
+          const genericItem = lastActivePathItem.slice(0, lastActivePathItem.lastIndexOf('-'));
+          if (this.matchSurroundingItems(lastChildrenUID, genericItem)) return;
+        }
+
+        // Set new active UID to the last matched item
+        // Note: if a match is not found, last matched ancestor is highlighted
+        this.setActiveUID(lastChildrenUID);
         return;
       }
-      // if there is an activeUID, unset it, as we probably navigated back to the root
+      // if there is an activeUID, but still no match found in tree
+      // unset it, as we probably navigated back to the root
       if (this.activeUID) {
         this.setActiveUID(null);
         return;
       }
       // Just track the open nodes, as setting the activeUID as null wont do anything.
       this.trackOpenNodes(this.nodeChangeDeps);
     },
+    /**
+     * try to match if any of the `siblings`,`children` or any of the `parents`
+     * of the UID match the item
+     */
+    matchSurroundingItems(UID, item) {
```

Comment:
Do you think we should rename this method, so its obvious it mutates the state? I dont really have a better name in mind... 

Also lets rename `item` to `path` as its a path.

Reply:
I can add a comment to make it obvious it mutates the state. Can't think of a better name either...